---
output:
  pdf_document: default
  html_document: default
---
# Final Report for Project 2
# Author: Shreya Das

# Introduction
This study is looking into the role of TYK2 gene in the development of beta pancreatic cells to understand its implication in Type I Diabetes. It is known that Type I Diabetes is a autoimmune disease where the immune system attacks and destroys pancreatic beta cells, which is a crucial cell population that produces insulin that is required to regulate blood sugar levels. Molecularly, it found that TYK2 acts in intracellular signal transducer and activator of transcription (STAT) signaling which is also stimulated by cytokines and interferon including IFN-I. IFN-I is important in Type I diabetes as it up-regulated the MHC Class I protein to target against the autoimmunity towards beta cells. It has also been shown that rare loss-of-function of TYK2 in the Japanese population has been correlated with an increase in diabetes susceptibility. This is why the authors are interested in understanding the underlying biological mechanisms and roles that TYK2 plays in the prognosis of Type I Diabetes.

The authors use RNA-Seq in particular in developing iPSC pancreatic beta cells to understand the changes in gene expression as a result of knocking out TYK2. By doing this, the authors can elucidate what genes are impacted by the loss of TYK2, but also what pathways are impacted by the loss of TYK2. Specifically by doing pathway analysis via KEGG and GSEA, the authors can understand what important roles TYK2 play in the development of Type I Diabetes and what potential treatment targets would be effective against Type I Diabetes.

# Methods

We used NextFlow to build the pipeline required to obtain the raw counts data required for DESeq2 analysis in R from the pair-end RNASeq reads of S5 stage EP cells in WT and TYK2-KO conditions. Our main goal was to compare the gene expression in WT and TYK2-KO at S5 stage EP cells.

### FASTQC

First, we read-in the paired-end read fasta files to evaluate the quality of the individual RNASeq reads using FastQC (v0.12.1). We used default parameters to generate FastQC reports for each the paired-end reads. 

### Generating Gene Ids file

Then we used a custom python script called GENERATEGENEIDS (pandas v2.2.3) to generate a file with all the gene names and associated gene ids from the annotated gtf file (Gencode Human Release v45 gtf annotation file.). Because this is custom python script the input is the gtf file and the output is the .csv file called "genes.csv". This will be important for our R analysis in DESeq2, DAVID and FGSEA analysis.

### STAR

Next, we generated a STAR index file using STAR (v2.7.11b) with the reference genome (GRCh38) and the gtf annotation file. Here we used the default parameters with defining the paths of the reads and the annotated gtf file and the --runThreadN parameter with 16 threads. 

Using NextFlow channeling we created a channel to contain the paired-end reads that had the sample names, and paths for read 1 and read 2. This channel was feed as the input for the STAR aligner.

We used STAR aligner (v2.7.11b) to align the the RNASeq reads (from the channel) to the STAR index file that was generated. The parameters that we used are the -runThreadN parameter with a value of 16 threads, the zcat option was for --readFilesCommand, the sample id name was used for --outFileNamePrefix, the BAM and SortedByCoordinate option was used for outSAMtype, and we redirected the output to a log output file.

### MultiQC Analysis

To do MultiQC analysis, we channeled all the outputs of the STAR_ALIGNER process and the FASTQC process into a channel.

Then we generated a MultiQC (v1.25) report using the outputs from the FastQC reports and the STAR Aligner reports to generate a single report on all of the reads from samples after aligning them to the reference genome. We evaluated the reads which we further analyzed in "QC Evaluation" For MultiQC we used the default parameters. 

### VERSE for quantification of counts for exons

Using VERSE (v0.1.5), we quantified the number of counts/reads for each exon. The parameters used were the -T parameter to define that 8 threads will be used for the process, -S parameter to indicate that we are using pair-end reads, -a with the gtf annotation file, and -o with the sample name and the input bam file from the STAR_ALIGNER process.

### Generating a counts data file for R analysis

We created a channel to grab all the .exon.txt files that were generated by VERSE to input into the custom python script in the next step.

Finally we used a custom python script called CONCAT_COUNTS (pandas v2.2.3) to output a .csv file with all the counts from each sample for all the exons in one file (from the channel in the previous step). Since this was a custom python script the parameters was the exon file outputs from VERSE and the output file would be called "counts.csv".

# QC Evaluation

In the MultiQC report, the number of reads range from 84.5 million to 118.9 million for all the samples.Additionally, FastQC raises warnings for the Per Base Sequence Content, Sequence Duplication Levels, and Overrepresented sequences by sample. 

### Per Base Sequence Content 

There is a bias in the beginning of the sequences. This can be attributed to using random hexamers in RNASeq libraries. While this is something that the FastQC report warned about, in our analysis it most likely will not severely impact the downstream analysis. 

### Sequence Duplication Levels

The plot in the FastQC report indicates a couple of peaks of sequence duplication. While in most cases this would be concerning, these sequence duplication rates may be attributed to the high-depth that the RNA-Seq experiment was performed at (200 million reads per sample), such that there is a higher likelihood that the same read would be generated in the experiment than with a lower-depth.

### Overrepresented Sequences

Looking at the overrepresented sequences that were determined by FastQC (and a quick search), some of these sequences are identified as Illumina Sequencer RNASeq adapters. This is sometimes an expected result if the adapters were not properly trimmed from the final reads before building the pipeline.

### Alignment Rate of reads

Looking at the STAR summary statistics, the rate of alignment from uniquely aligned reads to the reference genome ranges from 91.9% to 94.0%. For sufficient and good results, rates above 60-70% for alignment is expected, indicating that our alignment process went well.  

### Summary

Overall, we evaluate the quality of reads to be sufficient. Idealistically, performing trimming on the raw reads would remove the adapter sequences. However, for a task like differential gene expression this usually not required as STAR does a 'soft-trim' on the raw reads which would yield sufficient results. 

# Filtering the counts matrix
Before filtering, the number of genes from the raw counts data was 63241 genes. After filtering, the number of genes went down to 23387 genes.

### Load-in necessary libraries

Load the DESeq2 library and tidyverse into the markdown notebook.
```{r}
library(DESeq2)
library(tidyverse)
```

In order to the differential gene expression analysis, we will need to have a counts matrix (this was generated by the nextflow pipeline), the colData, and the design formula. In this notebook we will be generating the colData and the design formula.

### Read-in the raw counts with counts.csv and generating a counts matrix

This is the raw counts that were obtained for the nextflow pipeline. 
```{r}
counts <- as.matrix(read.csv("counts.csv", row.names='gene'))
counts
```
To get the number of genes that is for the raw counts data, we use the dim function on a matrix. We see that there are 63241 genes for the 6 samples.
```{r}
dim(counts)
```

Here we can see the rows of the counts matrix that was generated by our nextflow pipeline.

### Pre-filtering the raw counts
We need to pre-filter the raw counts so that we eliminate the genes that are not informative (i.e., have too low amount of counts). We can determine the threshold count depending on our data from counts.csv.

```{r}
smallestGroupSize <- 3
keep <- rowSums(counts >= 5) >= smallestGroupSize
filt_counts <- counts[keep,]
filt_counts
```
I choose this method for filtering the reads because I saw that most counts have less than 5 counts or more than 5 counts. So I choose 5 counts as my minimum threshold for the counts. The variable smallestGroupSize is the way that I also filtered the genes that have significant gene expression, such that at least 3 samples must have 5 or more counts to be kept in the filtered counts data. This corresponds to the fact that in our current study we have 3 replicates for each treatment.

```{r}
dim(filt_counts)
```
After filtering there is 23387 genes that are somewhat interesting for the 6 samples.

# Differential Experession Analysis

To see the table of top 10 differentially expressed genes provided by DESeq2 ranked by padj, go to the section called "Table of top 10 significant genes ranked by padj". 

We choose a p-value of 0.05, such that genes that were considered significantly differential expressed had a padj value of less than 0.05, regardless if the gene was up-regulated or down-regulated. At this p-value of 0.05, there was 1,227 genes that was considered significant. 

After doing ENRICHR analysis (the image is included in the directory under KEGG 2021...png) on the genes that were significantly expressed, it was observed that there was a enrichment for pathways in PI3K-Akt signalling, ECM-receptor interaction, Maturity onset diabetes of the young, GABAergic synapse, and p53 signaling pathway. Some of these pathways makes sense such as PI3K-Akt signalling and maturity onset diabetes of the young as both are implicated in diabetes. The first is the main insulin signalling pathway that regulates the metabolism of glucose (https://pubmed.ncbi.nlm.nih.gov/30263000/) and the latter is evident that our cell line is immature pancreatic cells that leads to diabetes (KEGG Entry# map04950). The other pathways like ECM-R interaction is related to diabetes, as the integrin receptors interact with ECM proteins to affect glucose uptake (https://pmc.ncbi.nlm.nih.gov/articles/PMC4490038/), GABAergic synapse pathways are affected as there are interlinks between glucose and GABA signalling to produce neuronal excitability and homeostatic state of glucose (https://pmc.ncbi.nlm.nih.gov/articles/PMC3198728/), and p53 signaling seems to promote insulin resistance and obesity (https://pmc.ncbi.nlm.nih.gov/articles/PMC5282448/). Overall, the top pathways that are enriched seems to make intuitive sense in the experiment's context.

The top up-regulated pathways (p = 0.05) enriched in fgsea analysis were MIKKELSEN_MEF_HCP_WITH_H3K27ME3, MIKKELSEN_MCV6_HCP_WITH_H3K27ME3, REACTOME_NEURONAL_SYSTEM, FISCHER_DIRECT_P53_TARGETS_META_ANALYSIS, and MEISSNER_NPC_HCP_WITH_H3K4ME2_AND_H3K27ME3. The first 2 and last pathways are involved in histone modifications which affects gene expression at the transcription level. These histone modifications may be important in differentiation of the cells and this specific mark is associated with gene repression. It could be that this pathway is upregulated at the S5 state in order to repress genes that are not associated with the next stage of development for these cells. The top down-regulated pathways (p = 0.05) enriched in fgsea analysis were DIRECT_P53_TARGETS_META_ANALYSIS, PER_BREAST_BASAL_VS_LUMINAL_UP, PID_P53_DOWNSTREAM_PATHWAY, SAUL_SEN_MAYO, and IGLESIAS_E2F_TARGETS_UP. The first and third pathways corresponds with the results from the ENRICHR analysis such that p53 pathways are significant. The SAUL_SEN_MAYO pathway enrichment makes some intuitive sense such that these cells are not suppose to to be showing senescence as they are very naive cells are in the beginning stages of development. For this pathway to be down-regulated makes sense in the context of the cellular model used. IGLESIAS_E2F_TARGETS_UP seems to be pathways involved in cell growth control especially in cancer. Additionally this pathway has been studied in the context of pancreatic cells, such that E2F receptors are important to control the growth of mature pancreatic cells and is required for the maintenance of differentiated pancreatic phenotype.

There was a few similar area of pathway enrichment which involved p53 in both ENRICHR and fgsea such presence of p53 and neuronal related pathways, both of which makes contextual sense in diabetes and glucose uptake regulation. However, we observe that the differences between the results of both analyses vary a lot more. Such that ENRICHR analysis is more specific on the molecular level focusing more on proteins or biological molecules and fgsea analysis focus on high-level pathways.

### Generating the colData

We need to manually create the colData.csv file which DESeq2 refers to understand what types of comparisons we want to make. Here, we want to make a comparison between the control and experimental replicates. With this .csv file and the design formula we will be able to make a comparison between the different types of sample (control and experiment).
```{r}
coldata <- read.csv("colData.csv")
coldata
```
### The Design Formula
Here, we are interested in the differential gene expression as a result of treatment either control or experiment. For each treatment condition we have 3 replicates, so we want to compare these replicates against each other. 

Our design formula will be:
~ treatment

### DESeq2 object
```{r}
dds <- DESeqDataSetFromMatrix(countData = filt_counts,
                              colData = coldata,
                              design = ~ treatment)
```

### Running DESeq

We will run DESeq on the object that we created.

```{r}
dds <- DESeq(dds)
res <- results(dds, contrast = c("treatment", "con", "exp"))
res
```

### Table of top 10 significant genes ranked by padj

We will generate a table that contains the top ranked genes according to padj value. The lowest padj value is at the top and increases as you go down the table. 
```{r}
resOrdered <- res[order(res$padj),]
deseq2_res <- as_tibble(resOrdered, rownames='gene')
deseq2_res
```
```{r}
top_10_genes <- head(deseq2_res, 10)
write.table(top_10_genes, "top_10_genes.txt", col.names = FALSE, row.names = FALSE, quote = FALSE)

top_10_genes
```

### Joining Gene names to the DESeq table
Load the readr library to be able to read in the csv file.
```{r}
library (readr)
```
```{r}
ids <- read_tsv(("genes.csv"), col_names = c("gene", "symbol"))

ids
```
Joining the deseq2 object with the .csv file with the gene names.
```{r}
deseq2_res <- deseq2_res %>% left_join(ids) %>% relocate(symbol)

deseq2_res
```
# ENRICHR Analysis
Based on what was given in the paper, the authors used a FDR of < 0.01 to filter for significantly expressed genes for both up-regulated and down-regulated. Here we will be using a padj value of < 0.05 and a log2FoldChange of greater than 0 or less than 0 to get the significantly expressed genes that were up-regulated and down-regulated.

```{r}
filtered_deseq2 <- deseq2_res %>% filter(padj <.05 & (log2FoldChange > 0 | log2FoldChange < 0)) %>% select(symbol)

write.table(filtered_deseq2, "interesting_genes.txt", col.names = FALSE, row.names = FALSE, quote = FALSE)

filtered_deseq2
```

### GSEA Analysis

We will perform GSEA Analysis using log2FoldChange as the ranking metric.

```{r}
fgseaOrdered <- deseq2_res %>% arrange(desc(log2FoldChange)) %>% drop_na(log2FoldChange) %>% distinct(symbol, .keep_all = TRUE)

rnk_list <- setNames(fgseaOrdered$log2FoldChange, fgseaOrdered$symbol)

```

Read in the GMT pathways file
```{r}
library('fgsea')
```
```{r}

pathways <- gmtPathways("c2.all.v2025.1.Hs.symbols.gmt")
head(pathways)
```

### Perform the FGSEA Analysis

```{r}
fgseaRes <- fgsea(pathways = pathways, 
                  stats = rnk_list,
                  minSize = 15,
                  maxSize = 500)
```

```{r}
head(fgseaRes[order(pval), ])
```

```{r}
topPathwaysUp <- fgseaRes[ES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- fgseaRes[ES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
plotGseaTable(pathways[topPathways], rnk_list, fgseaRes, 
              gseaParam=0.5)
```

## RNAseq Quality Control

The PCA plot shows a good clustering for the control samples, which is a good confirmation that the 3 replicates show similar reads from the RNASeq experiment. However, when analyzing the the experimental samples there is 1 replicate that is separate from the clustering of the 2 other replicates. There could be a multitude of reasons why there is this variation such as batch effects, biological factors, or simply when the experiment was performed for each of these replicates. Considering that this is a high-depth RNA-seq experiment, this shouldn't affect the results of the differential expression analysis and pathway enrichment too much, however ideally this is effect that we would try to remove as much as possible.

Overall, the heatmap shows that the corresponding samples have the shortest distance in terms sample-to-sample distances (i.e., con-con and exp-exp); this is expected and a good confirmation that the RNASeq experiment was performed correctly. However, the hierarchical clustering indicates that one of the experimental replicates (exp_rep_3) has close enough distance with the control replicates. This result also corresponds with the results of the PCA plot such that one the experimental replicates didn't cluster with the other experimental replicates. 

Ideally, we would try to repeat the experiment to ensure that the replicates do cluster similarly using both the PCA plot and heatmap clustering too. This could indicate that there were batch effect, biological factors, and technical variation.

### Normalization the counts matrix
```{r}
vsd <- vst(dds, blind=FALSE)

head(assay(vsd), 5)
```
### PCA Plots on normalized counts matrix
```{r}
plotPCA(vsd, intgroup=c("treatment"))
```
### Create a heatmap of sample-to-sample distances

```{r}
sampleDists <- dist(t(assay(vsd)))

library(pheatmap)
library("RColorBrewer")

sampleDistMatrix <- as.matrix(sampleDists)
colors <- colorRampPalette( rev(brewer.pal(9, "Reds")) )(255)
pheatmap(sampleDistMatrix, col=colors, labRow = vsd$treatment, labCol = vsd$treatment)
```
### Replication of figures

Using the author's significance level of FDR < 0.01, we found 408 up-regulated genes and 323 down-regulated genes. The results obtained by the authors indicate an observation of 319 up-regulated and 412 down-regulated genes. This difference may be attributed to the pre-filtering method for the raw counts data that was used. The authors didn't indicate what method was used for the analysis.

In terms of pathway enrichment, the authors had observed an enrichment for Extracellular matrix organization, Integrin cell surface interactions, ECM proteoglycans, and Signalling by Receptor Tyrosine Kinases for up-regulated and Regulation of beta-cell development, Neuronal System, and Regulation of gene expression in beta cells. We had observed an enrichment for Fischer Direct P53 Targets Meta Analysis, Saul Sen Mayo, Pid P53 Downstream Pathway, Macaeva Pbmc Response To Ir, and Iglesias E2f Targets Up for up-regulated pathways and Mikkelsen Mef Hcp With H3k27me3, Martoriati Mdm4 Targets Neuroepithelium Dn, Mikkelsen Mcv6 Hcp With H3k27me3, Reactome Neuronal System, and Meissner Npc Hcp With H3k4me2 and H3k27me3 for down-regulated pathways. For the author's findings, pathways involved in structure were upregulated and we observed that pathways involved in p53 were up-regulated. Additionally, the author's observed that pathways involved in beta cell development and regulation were down-regulated and we observed down-regulation in pathways involved in methylation marks which could be similar to regulation of gene expression in beta cells, similar to the authors. Some of these differences could be attributed to the authors using a different enrichment tool (Reactome vs. FGSEA) and a different gmt file with pathways to provide in the enrichment process.

## Recreating Figure 3C
```{r}
# Load required libraries
library(ggplot2)
library(ggrepel)  # for non-overlapping text labels

# Create your data frame with your differentially expressed genes
# Replace this with your actual data
df <- as.data.frame(deseq2_res)

# Calculate -log10(p-value)
df$neg_log10_pval <- -log10(df$pvalue)

# Define significance thresholds
log2FC_threshold <- 1  # absolute log2 fold change threshold
pval_threshold <- 0.01  # p-value threshold

# Classify genes as upregulated, downregulated, or non-significant
df$regulation <- ifelse(df$log2FoldChange > log2FC_threshold & df$pvalue < pval_threshold, "Upregulated",
                        ifelse(df$log2FoldChange < -log2FC_threshold & df$pvalue < pval_threshold, "Downregulated",
                               "Non-significant"))

# Define colors matching the paper (soft red for up, iris blue for down, black for NS)
colors <- c("Upregulated" = "#E57373", 
            "Downregulated" = "#64B5F6", 
            "Non-significant" = "black")

top_genes <- df %>%
  filter(
    padj < 0.001,  # Very significant
    abs(log2FoldChange) > 2  # Large fold change
  ) %>%
  arrange(padj)

# Create volcano plot
volcano_plot <- ggplot(df, aes(x = log2FoldChange, y = neg_log10_pval)) +
  geom_point(aes(color = regulation), alpha = 0.6, size = 2) +
  scale_color_manual(values = colors) +
  
  geom_text_repel(
    data = top_genes,
    aes(label = symbol),
    size = 3,
    box.padding = 0.5,
    point.padding = 0.3,
    max.overlaps = 30,
    segment.color = "grey50"
  ) +
  
  # Add threshold lines (optional)
  geom_hline(yintercept = -log10(pval_threshold), linetype = "dashed", 
             color = "grey50", linewidth = 0.5) +
  geom_vline(xintercept = c(-log2FC_threshold, log2FC_threshold), 
             linetype = "dashed", color = "grey50", linewidth = 0.5) +
  
  # Customize theme
  theme_classic() +
  theme(
    legend.position = "right",
    legend.title = element_blank(),
    axis.text = element_text(size = 10, color = "black"),
    axis.title = element_text(size = 11, face = "bold"),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) +
  
  # Axis labels
  labs(
    x = expression(bold("Expression difference (log"[2]*" FC)")),
    y = expression(bold("Significance (-log"[10]*" p value)"))
  ) +
  
  # Set axis limits (adjust based on your data range)
  coord_cartesian(xlim = c(min(df$log2FoldChange) - 0.5, max(df$log2FoldChange) + 0.5),
                  ylim = c(0, 15))

# Display the plot
print(volcano_plot)

# Save the plot
ggsave("volcano_plot_figure3c.pdf", volcano_plot, 
       width = 6, height = 5, units = "in", dpi = 300)

```
### Recreating Figure 3F
```{r}
library(ggplot2)
library(dplyr)
library(stringr)
library(fgsea)

# Assuming 'fgsea_results' is your fgsea output
# fgsea_results <- fgsea(pathways = pathways, stats = ranked_genes, ...)

# Prepare fgsea results for plotting
gsea_plot_data <- fgseaRes %>%
  as.data.frame() %>%
  
  # Filter for significant pathways
  filter(padj < 0.05) %>%
  
  # Calculate percentage: size of pathway overlap / total pathway size * 100
  mutate(
    percentage = (size / length(unique(unlist(pathway)))) * 100,
    # OR simpler: just use size directly or calculate from leadingEdge
    # percentage = (sapply(leadingEdge, length) / size) * 100,
    
    # Clean pathway names
    pathway_clean = str_replace(pathway, "^.*~", ""),
    pathway_clean = str_replace(pathway_clean, "^HALLMARK_", ""),
    pathway_clean = str_replace(pathway_clean, "^GO_", ""),
    pathway_clean = str_replace(pathway_clean, "^KEGG_", ""),
    pathway_clean = str_replace_all(pathway_clean, "_", " "),
    pathway_clean = str_to_title(pathway_clean),
    pathway_clean = str_wrap(pathway_clean, width = 50),
    
    # Determine regulation based on NES (Normalized Enrichment Score)
    regulation = ifelse(NES > 0, "UP", "DOWN"),
    
    # Calculate -log10(padj) for coloring by significance
    neg_log10_padj = -log10(padj)
  ) %>%
  
  # Select top pathways from each direction
  group_by(regulation) %>%
  arrange(padj) %>%
  slice_head(n = 5) %>%  # Top 5 from each direction
  ungroup() %>%
  
  # Order for plotting
  arrange(regulation, desc(abs(NES)))

# Set factor levels to control plot order
gsea_plot_data$pathway_clean <- factor(gsea_plot_data$pathway_clean, 
                                       levels = rev(gsea_plot_data$pathway_clean))

# Define colors
colors <- c("UP" = "#E57373", "DOWN" = "#64B5F6")

# Create the plot using NES (common for fgsea) or percentage
# Option 1: Using percentage
plot1 <- ggplot(gsea_plot_data, aes(x = percentage, y = pathway_clean, fill = regulation)) +
  geom_bar(stat = "identity", width = 0.7) +
  scale_fill_manual(values = colors,
                    labels = c("DOWN" = "Downregulated", "UP" = "Upregulated")) +
  theme_classic() +
  theme(
    axis.text.y = element_text(size = 9, color = "black"),
    axis.text.x = element_text(size = 9, color = "black"),
    axis.title.x = element_text(size = 10, face = "bold"),
    axis.title.y = element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1),
    panel.grid.major.x = element_line(color = "grey90", linewidth = 0.5)
  ) +
  labs(x = "Percentage of DE genes in category /\nall genes in category (%)")+
  coord_cartesian(xlim = c(0, max(gsea_plot_data$percentage) * 1.5))

# Option 2: Using NES (Normalized Enrichment Score) - more common for GSEA
plot2 <- ggplot(gsea_plot_data, aes(x = NES, y = pathway_clean, fill = regulation)) +
  geom_bar(stat = "identity", width = 0.7) +
  scale_fill_manual(values = colors,
                    labels = c("DOWN" = "Downregulated", "UP" = "Upregulated")) +
  theme_classic() +
  theme(
    axis.text.y = element_text(size = 9, color = "black"),
    axis.text.x = element_text(size = 9, color = "black"),
    axis.title.x = element_text(size = 10, face = "bold"),
    axis.title.y = element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1),
    panel.grid.major.x = element_line(color = "grey90", linewidth = 0.5)
  ) +
  labs(x = "Normalized Enrichment Score (NES)") +
  geom_vline(xintercept = 0, linetype = "dashed", color = "grey50")

# Option 3: Using -log10(padj) colored by direction
plot3 <- ggplot(gsea_plot_data, aes(x = neg_log10_padj, y = pathway_clean, fill = regulation)) +
  geom_bar(stat = "identity", width = 0.7) +
  scale_fill_manual(values = colors,
                    labels = c("DOWN" = "Downregulated", "UP" = "Upregulated")) +
  theme_classic() +
  theme(
    axis.text.y = element_text(size = 9, color = "black"),
    axis.text.x = element_text(size = 9, color = "black"),
    axis.title.x = element_text(size = 10, face = "bold"),
    axis.title.y = element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1)
  ) +
  labs(x = "-log10(adjusted p-value)")

# Display plots
print(plot1)  # Percentage version
print(plot2)  # NES version (recommended for GSEA)
print(plot3)  # Significance version


```

### Comparision of number of up and down regulated genes

```{r}
filtered_deseq2_up <- deseq2_res %>% filter(padj <.01 & (log2FoldChange > 0)) %>% select(symbol)

filtered_deseq2_down <- deseq2_res %>% filter(padj <.01 & (log2FoldChange < 0)) %>% select(symbol)

dim(filtered_deseq2_up)
```
```{r}
dim(filtered_deseq2_down)
```


## SessionInfo

The sessioninfo function is one way we can attempt to be transparent about our
environment management in R.

This will print out a summary of the current packages installed in the environment.

```{r}
sessioninfo::session_info()
```
